src/checkout-buttons/strategies/googlepay/googlepay-stripe-button-strategy.spec.ts

```diff
@@ -107,108 +99,72 @@ describe('GooglePayCheckoutButtonStrategy', () => {
             await strategy.initialize(checkoutButtonOptions);
 
-            strategy.deinitialize();
+            await strategy.deinitialize();
 
             expect(paymentProcessor.deinitialize).toBeCalled();
         });
 
-        it('succesfully deinitializes the strategy', async () => {
+        it('removes the child elements from attached to the containerId', async () => {
             await strategy.initialize(checkoutButtonOptions);
 
-            strategy.deinitialize();
-
-            if (checkoutButtonOptions.containerId) {
-                const button = document.getElementById(checkoutButtonOptions.containerId);
+            const button = document.getElementById(containerId);
 
-                if (button) {
-                    expect(button.firstChild).toBe(null);
-                }
-            }
+            expect(button).toHaveProperty('firstChild', walletButton);
 
-            container = document.createElement('div');
-            document.body.appendChild(container);
-        });
-
-        it('Validates if strategy is loaded before call deinitialize', async () => {
             await strategy.deinitialize();
 
-            if (checkoutButtonOptions.containerId) {
-                const button = document.getElementById(checkoutButtonOptions.containerId);
-
-                if (button) {
-                    expect(button.firstChild).toBe(null);
-                }
-            }
+            expect(button).toHaveProperty('firstChild', null);
+        });
 
-            container = document.createElement('div');
-            document.body.appendChild(container);
+        it('fails silently if strategy was not initialized', () => {
+            expect(async () => await strategy.deinitialize()).not.toThrow();
         });
     });
 
     describe('#handleWalletButtonClick', () => {
-        let googlePayOptions: CheckoutButtonInitializeOptions;
-        let paymentData: GooglePaymentData;
-
-        beforeEach(() => {
-            googlePayOptions = getCheckoutButtonOptions(Mode.GooglePayStripe);
-
-            paymentData = getGooglePaymentDataMock();
-        });
-
         it('handles wallet button event', async () => {
-            jest.spyOn(paymentProcessor, 'displayWallet').mockReturnValue(Promise.resolve(paymentData));
+            checkoutButtonOptions = getCheckoutButtonOptions(Mode.GooglePayStripe);
+
+            jest.spyOn(paymentProcessor, 'displayWallet').mockReturnValue(Promise.resolve(getGooglePaymentDataMock()));
             jest.spyOn(paymentProcessor, 'handleSuccess').mockReturnValue(Promise.resolve());
             jest.spyOn(paymentProcessor, 'updateShippingAddress').mockReturnValue(Promise.resolve());
 
-            await strategy.initialize(googlePayOptions).then(() => {
-                walletButton.click();
-            });
+            await strategy.initialize(checkoutButtonOptions);
+
+            walletButton.click();
```

Comment:
same as in the other test, what's the difference when walletButton is provided? is it that `paymentProcessor.initialize` is only called after the wallet is clicked? If so, might be better to be explicit about it:

```js
await strategy.initialize(checkoutButtonOptions);
expect(paymentProcessor.initialize).not.toHaveBeenCalled();
walletButton.click();
expect(paymentProcessor.initialize).toHaveBeenCalledWith(/** ...arg */);
```